Userland/Libraries/LibJS/Runtime/Reference.cpp

```diff
@@ -13,53 +13,38 @@
 namespace JS {
 
 // 6.2.4.6 PutValue ( V, W ), https://tc39.es/ecma262/#sec-putvalue
-void Reference::put_value(GlobalObject& global_object, Value value)
+ThrowCompletionOr<void> Reference::put_value(GlobalObject& global_object, Value value)
 {
     auto& vm = global_object.vm();
 
-    if (!is_valid_reference()) {
-        vm.throw_exception<ReferenceError>(global_object, ErrorType::InvalidLeftHandAssignment);
-        return;
-    }
+    if (!is_valid_reference())
+        return vm.throw_completion<ReferenceError>(global_object, ErrorType::InvalidLeftHandAssignment);
 
     if (is_unresolvable()) {
-        if (m_strict) {
-            (void)throw_reference_error(global_object);
-            return;
-        }
-        MUST(global_object.set(m_name, value, Object::ShouldThrowExceptions::No));
-        return;
+        if (m_strict)
+            return throw_reference_error(global_object);
+        TRY(global_object.set(m_name, value, Object::ShouldThrowExceptions::No));
```

Comment:
Why does this change from MUST to TRY?

Reply:
Because the specification says so, so I did a drive-by, can make it a separate commit if you prefer.

Reply:
fine by me, thanks for clarifying 👍